File: 0643-maximum-average-subarray-i/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        avg = 0
        for i in range(k):
            avg += nums[i]
        maximum = float(avg) / k
        for i in range(k,len(nums)):
            avg = avg+nums[i] - nums[i-k]
            current_avg = float(avg) / k
            maximum = max(maximum,current_avg)
        return maximum    
            

        

File: 0643-maximum-average-subarray-i/solution.c
double findMaxAverage(int* nums, int numsSize, int k) {
    if (numsSize == 0 || k <= 0 || k > numsSize) {
        return 0; // Handle edge cases
    }

    double sum = 0;
    for (int i = 0; i < k; i++) {
        sum += nums[i]; // Initialize the sum with the first k elements
    }

    double max = sum; // Set max to the average of the first k elements

    for (int i = k; i < numsSize; i++) {
        sum += nums[i] - nums[i - k]; // Slide the window
        if (sum > max) {
            max = sum; // Update max if current sum is greater
        }
    }

    return max / k; // Return the maximum average
}

